core/camel-core-processor/src/main/java/org/apache/camel/processor/Splitter.java

```diff
@@ -203,6 +205,8 @@ public Iterator<ProcessorExchangePair> iterator() {
                 private int index;
                 private boolean closed;
 
+                private final Map<String, Object> data = new ConcurrentHashMap<>();
```

Comment:
You should only create this map if its transacted (otherwise its waste of cpu for regular non transacted). And the name data should be txData to better indicate its for transaction

Reply:
yeah, I will update it.